### PR TITLE
(DOCSP-13959): cross-namespace deployment docs

### LINKED ISSUE #316: [Documentation] Cross-namespace deployment example
Version 0.4.0 of the operator introduced support for cross-namespace deployments. Would it be possible to have documentation and/or a working example ?

I tried to do it by installing the ClusterRole/ClusterRoleBinding and defining `WATCH_NAMESPACE="*"`, to no avail.

### PR DESCRIPTION
**Pull request by jwilliams-mongo**
### All Submissions:

* [x] Have you opened an Issue before filing this PR?
* [x] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* [x] Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if such).

closes #316 

[JIRA](https://jira.mongodb.org/browse/DOCSP-13959)

### CODE REVIEW

File: docs/install-upgrade.md [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 - [Install the Operator](#install-the-operator)
   - [Prerequisites](#prerequisites)
+  - [Understand MongoDB Community Kubernetes Operator Deployment Scopes](#understand-mongodb-community-operator-deployment-scopes)
```
Comment (jeff-allen-mongo) [inline docs/install-upgrade.md]: When I click on this link I don't get directed to the new section. Is something off here?

File: docs/install-upgrade.md [hunk 2]
```diff
@@ -20,6 +21,58 @@ Before you install the MongoDB Community Kubernetes Operator, you must:
    ```
    git clone https://github.com/mongodb/mongodb-kubernetes-operator.git
    ```
+4. Review the possible Operator [deployment scopes](#understand-mongodb-community-operator-deployment-scopes) and configure the Operator to watch other namespaces, if necessary.
```
Comment (jeff-allen-mongo) [inline docs/install-upgrade.md:24]: Same comment here. "deployment scopes" doesn't seem to link properly.

File: docs/install-upgrade.md [hunk 3]
```diff
@@ -20,6 +21,58 @@ Before you install the MongoDB Community Kubernetes Operator, you must:
...
+
+### Understand MongoDB Community Kubernetes Operator Deployment Scopes
+
+You can deploy the MongoDB Community Kubernetes Operator with different scopes based on where you want to deploy Ops Manager and MongoDB Kubernetes resources:
+
+- [Operator in Same Namespace as Resources](#operator-in-same-namespace-as-resources)
+- [Operator in Different Namespace Than Resources](#operator-in-different-namespace-than-resources)
+- [Cluster-Wide Scope](#cluster-wide-scope)
```
Comment (jeff-allen-mongo) [inline docs/install-upgrade.md:32]: Cluster-wide scope link not working. The other two work fine.

**Review by jeff-allen-mongo — COMMENTED** [review]
Some of the links don't seem to work. Content looks solid though.

File: docs/install-upgrade.md [hunk 4]
```diff
@@ -20,6 +21,58 @@ Before you install the MongoDB Community Kubernetes Operator, you must:
...
+   ```yaml
+       spec:
+         containers:
+           - name: mongodb-kubernetes-operator
+             image: quay.io/mongodb/mongodb-kubernetes-operator:0.5.0
+             command:
+               - mongodb-kubernetes-operator
+             imagePullPolicy: Always
+             env:
+               - name: WATCH_NAMESPACE
```
Comment (jeff-allen-mongo) [inline docs/install-upgrade.md:59]: Can you highlight lines in markup? Might be helpful here if so, but definitely not necessary.

File: docs/install-upgrade.md (hunk 2 above)
Comment (jwilliams-mongo) [reply docs/install-upgrade.md:24]: fixed

**Review by jeff-allen-mongo — APPROVED** [review]
LGTM with one non-blocking comment.

File: docs/install-upgrade.md (hunk 4 above)
Comment (jwilliams-mongo) [reply docs/install-upgrade.md:59]: I don't think there's a simple way to do this. I did a little research but came up empty.

File: docs/install-upgrade.md [hunk 5]
```diff
@@ -20,6 +21,57 @@ Before you install the MongoDB Community Kubernetes Operator, you must:
    ```
    git clone https://github.com/mongodb/mongodb-kubernetes-operator.git
    ```
+4. Review the possible Operator [deployment scopes](#understand-mongodb-community-operator-deployment-scopes) and configure the Operator to watch other namespaces, if necessary.
+
+### Understand Deployment Scopes
+
+You can deploy the MongoDB Community Kubernetes Operator with different scopes based on where you want to deploy Ops Manager and MongoDB Kubernetes resources:
```
Comment (bznein) [inline docs/install-upgrade.md:28]: The Community Operator doesn't support Ops Manager :)

Comment (alyacb) [reply docs/install-upgrade.md:28]: +1

**Review by alyacb — CHANGES_REQUESTED** [review]
Looks good, but the reference to Ops Manager should be removed before merging.

File: docs/install-upgrade.md (hunk 5 above)
Comment (jwilliams-mongo) [reply docs/install-upgrade.md]: oof. copy / paste mistake from the enterprise operator. fixed.

**Review by alyacb — APPROVED** [review]
LGTM! 🎉

**Pull request merged**